Meme/Meme/ViewModel.swift

```diff
@@ -0,0 +1,63 @@
+//
+//  ViewModel.swift
+//  Meme
+//
+//  Created by Mesfin Bekele Mekonnen on 7/30/19.
+//  Copyright © 2019 Mesfin Bekele Mekonnen. All rights reserved.
+//
+
+import Foundation
+
+class ViewModel {
+
+    let networkClient: NetworkClient
+    private var didDownloadTemplates: (([Template]) -> Void)?
+    private var didDownloadTemplateExample: ((TemplateExample) -> Void)?
+    private var didDownloadMeme:((Meme) -> Void)?
```

Comment:
I think you can drop the delegates in favor or just using callbacks directly.

Reply:
Yeah, I was just using callbacks, and I read this article about weakifying self https://medium.com/anysuggestion/preventing-memory-leaks-with-swift-compile-time-safety-49b845df4dc6

then opted for using these delegates. When we set the delegate her, we weakify it so that the caller of the API (ViewController) doesn't have to remember to do it.